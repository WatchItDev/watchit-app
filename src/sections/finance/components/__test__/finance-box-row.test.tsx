import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import FinanceBoxRow from '../finance-box-row';

const renderComponent = () => {
  return render(
    <FinanceBoxRow>
      <div>Child 1</div>
      <div>Child 2</div>
    </FinanceBoxRow>,
  );
};
describe('[COMPONENTS] <FinanceBoxRow />', () => {
  it('renders correctly with children and matches snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('renders children correctly', () => {
    const { getByText } = renderComponent();
    expect(getByText('Child 1')).toBeInTheDocument();
    expect(getByText('Child 2')).toBeInTheDocument();
  });

  it('has correct styles', () => {
    const { container } = renderComponent();
    const boxElement = container.querySelector('div');
    expect(boxElement).toHaveStyle('display: flex');
    expect(boxElement).toHaveStyle('align-items: center');
    expect(boxElement).toHaveStyle('justify-content: space-between');
    expect(boxElement).toHaveStyle('margin-top: 4px');
    expect(boxElement).toHaveStyle('margin-bottom: 4px');
  });
});
