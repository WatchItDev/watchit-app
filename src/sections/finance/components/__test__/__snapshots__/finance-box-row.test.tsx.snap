// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[COMPONENTS] <FinanceBoxRow /> > renders correctly with children and matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1us6dsi"
  >
    <div>
      Child 1
    </div>
    <div>
      Child 2
    </div>
  </div>
</div>
`;

exports[`[COMPONENTS] <FinanceBoxRow /> > to match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1us6dsi"
  />
</div>
`;
